### DISCUSSION TITLE
Uniqueness of the generated id

### DISCUSSION THREAD

**Discussion by d3bt3ch** [sqids/sqids-java#18] [General]
Given `unique long integers` would `sqids` always create `unique id` without any `collision or overlap` ?

**Comment by 0x3333** [comment]
Yes, if you keep the parameters the same.

**Comment by d3bt3ch** [comment]
Thanks. I missed the FAQ where your reply is clearly mentioned. Another concern is from the FAQ `decoding random IDs can sometimes produce the same numbers`. What are the odds because we are planning to use `sqids` to stringify our long ids in json output. If two ids produce the same long then it could be a problem.

**Reply by 0x3333** [reply]
If the IDs you generated are using the same parameters, they can't collide, they are unique.

The FAQ is regarding a random generated ID, like brute-force, etc. The algorithm is not cryptographically secure.
